src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1722,6 +1723,45 @@ def _public_key_bytes(self, encoding, format, evp_pkey, cdata):
                 write_bio = self._lib.i2d_RSAPublicKey_bio
 
             key = cdata
+        elif format is serialization.PublicFormat.OpenSSH:
+            if encoding is not serialization.Encoding.OpenSSH:
+                raise ValueError(
+                    "OpenSSH format must be used with OpenSSH encoding"
+                )
+            if isinstance(key, rsa.RSAPublicKey):
+                public_numbers = key.public_numbers()
+                return b"ssh-rsa " + base64.b64encode(
+                    serialization._ssh_write_string(b"ssh-rsa") +
+                    serialization._ssh_write_mpint(public_numbers.e) +
+                    serialization._ssh_write_mpint(public_numbers.n)
+                )
+            elif isinstance(key, dsa.DSAPublicKey):
+                public_numbers = key.public_numbers()
+                parameter_numbers = public_numbers.parameter_numbers
+                return b"ssh-dss " + base64.b64encode(
+                    serialization._ssh_write_string(b"ssh-dss") +
+                    serialization._ssh_write_mpint(parameter_numbers.p) +
+                    serialization._ssh_write_mpint(parameter_numbers.q) +
+                    serialization._ssh_write_mpint(parameter_numbers.g) +
+                    serialization._ssh_write_mpint(public_numbers.y)
+                )
+            else:
+                assert isinstance(key, ec.EllipticCurvePublicKey)
+                public_numbers = key.public_numbers()
+                curve_name = {
+                    ec.SECP256R1: b"nistp256",
+                    ec.SECP384R1: b"nistp384",
+                    ec.SECP521R1: b"nistp521",
+                }[type(public_numbers.curve)]
```

Comment:
Any curve outside of the 3 supported will raise a KeyError right now. This is probably not what we want.